dulwich/cli.py

```diff
@@ -729,6 +733,61 @@ def run(self, args) -> None:
             print(canonical_identity)
 
 
+class cmd_branch(Command):
+    def run(self, args) -> None:
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "branch",
+            type=str,
+            help="Name of the branch",
+        )
+        parser.add_argument(
+            "-d",
+            "--delete",
+            action="store_true",
+            help="Delete branch",
+        )
+        args = parser.parse_args(args)
+        if not args.branch:
+            print("Usage: dulwich branch [-d] BRANCH_NAME")
+            sys.exit(1)
+
+        r = Repo(".")
```

Comment:
Explicitly opening here isn't necessary, porcelain.branch_delete() and branch_create() accept a path

Reply:
```suggestion
```